faq.md

```diff
@@ -1,45 +1,634 @@
+    public Vector2i? Size { get; set; }            // Client/content size, in pixels
+    
+    public WindowBorder WindowBorder { get; set; } // Fixed, Resizable, Hidden (no bord.)
+    public WindowState WindowState { get; set; }   // Normal, Minim., Maxim., Hidden
+    public bool IsFullscreen { get; set; }         // Like Maximized, only bigger :-)
+}
+```
+
+A number of these, like `Location` and `Size` and `WindowState` and `IsFullscreen`, are re-exposed after the window is constructed as properties on the window itself, so you can use the equivalent window properties to update your window’s appearance and state as well.
+
+If you want to delay your window’s position/layout until after it has been constructed and important data for it has been loaded (like its position), a common pattern is something like this:
+
+```c#
+var nativeSettings = new NativeWindowSettings { StartVisible = false };
+using (var myWindow = new MyWindow(nativeSettings, gameSettings))
+{
+    myWindow.Run();
+}
+
+...
+    
+public class MyWindow : GameWindow
+{
+    ...
+        
+    public override OnLoad()
+    {
+        // Do setup...
+        ...
+            
+        // Now show it.
+        CenterWindow();
+        WindowState = WindowState.Normal;
+    }
+}
+```
+
+(Note that OpenTK 3 used a very different technique for setting up its initial window state:  Window parameters are passed as part of the `GameWindow` or `NativeWindow` constructor; there is no such thing as `NativeWindowSettings`; and configuration options are fairly limited.)
+
+### What’s up with OpenTK’s matrices?
+
+For historical reasons, OpenTK’s built-in `Matrix` structs store their data in row-major order, not column-major order.  This provides consistent backward-compatibility with existing software that uses OpenTK, but it can be confusing, for two reasons:
+
+- First, if you copy other code found online for OpenGL, you may need to *transpose* the matrices their code uses to get the expected behavior.
+- Second, matrix-matrix multiplications typically need to be done in _reverse_ order of those of most other libraries.
+
+This is a common source of confusion when using OpenTK, since it often prevents “code found on the Internet” from working directly.  We’re sorry if it causes you trouble, but it’s easy to work around it, and because so much existing software depends on it, it’s not something we plan to change any time soon.
+
+### What’s the difference between NativeWindow and GameWindow?
+
+OpenTK includes two classes, `NativeWindow` and `GameWindow`, for opening and displaying OpenGL windows.  What’s the difference between these two classes?
+
+- `NativeWindow` is the lowest-level class.  It includes the bare minimum of required mechanics to open and display a window, and not much else.  It’s small, light, and fast, and good for situations where you don’t need an update/render loop or you intend to write most of the update/render logic yourself.
```

Comment:
`NativeWindow` is not the lowest-level thing in OpenTK 4 (`GLFW.` is), and the split between `NativeWindow` and `GameWindow` isn't as clear as we would like it to be, but this explanation is probably ok.

Reply:
The split isn't always clear, but that's why it's followed up with `you probably want to use`.

I've added this addendum to the first bullet point:

> `NativeWindow` is the lowest-level class.  It includes the bare minimum of required mechanics to open and display a window, and not much else.  It’s small, light, and fast, and good for situations where you don’t need an update/render loop or you intend to write most of the update/render logic yourself.  (Note that `NativeWindow` is not the lowest *level*, merely the lowest level in managed .NET:  `NativeWindow` wraps native functionality provided by GLFW.)